components/page.jsx

```diff
@@ -10,8 +10,23 @@ var DevRibbon = (process.env.NODE_ENV === 'production' &&
                  process.env.SHOW_DEV_RIBBON !== 'on')
                 ? null
                 : require('./dev-ribbon.jsx');
+var config = require('../lib/config');
 
 var Page = React.createClass({
+  statics: {
+    handlerForPage: function(router, url) {
+      return router.match(url).routes[1].handler;
```

Comment:
Ah yeah unfortunately i actually had to dig into the source code to find that--it was really hard trying to figure out how to access route information outside of a component!

Two options:

1. We can submit docs for that method to react-router, which will hopefully make it a "public" method rather than potentially a private implementation detail.
2. We just use it and an exception will be thrown or a warning logged if we upgrade react-router and things break.

This message was typed with haste on a tiny keyboard with no tactile feedback.

> On Apr 23, 2015, at 5:16 PM, Mavis Ou <notifications@github.com> wrote:
> 
> In components/page.jsx:
> 
> >  
> >  var Page = React.createClass({
> > +  statics: {
> > +    handlerForPage: function(router, url) {
> > +      return router.match(url).routes[1].handler;
> hey @toolness can you point me to the docs for .match method? I could probably guess what that is for but I wanted to read through the docs to get a more clear sense. I couldn't find it anywhere on https://github.com/rackt/react-router/tree/master/docs 
> 
> —
> Reply to this email directly or view it on GitHub.
>